### PR TITLE
feat: improve upstream i18n

### LINKED ISSUE #1094: support full i18n
# Feature request

## Please describe your feature

**Checklist**
- [ ] Route
- [ ] Service
- [ ] Upstream
- [ ] Consumer
- [ ] SSL
- [ ] Plugin
- [ ] Other: Login, Logout, etc.

### PR DESCRIPTION
**Pull request by guoqqqi**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

- Related issues
resolve #1094

### CODE REVIEW

File: web/src/pages/Upstream/locales/en-US.ts [hunk 1]
```diff
@@ -61,10 +68,12 @@ export default {
   'upstream.step.healthy.checks.successes': 'Successes',
   'upstream.step.input.healthy.checks.successes': 'Please input successes',
   'upstream.step.healthy.checks.http_failures': 'HttpFailures',
+  'upstream.step.healthy.checks.active.create.req_headers': 'Creat req_headers',
```
Comment (juzhiyuan) [inline web/src/pages/Upstream/locales/en-US.ts:71]: Hi @LiteSun, please help @guoqqqi to update those i18n values when you are free 💯

**Comment by juzhiyuan** [comment]
image

@guoqqqi After you update the i18n, you need to update test cases in `/cypress` directory, please see snapshot.

**Comment by juzhiyuan** [comment]
image

**Comment by juzhiyuan** [comment]
ping @LiteSun @liuxiran

File: web/src/locales/en-US/menu.ts [hunk 2]
```diff
@@ -58,6 +58,7 @@ export default {
   'menu.account': 'Account',
   'menu.account.center': 'Account Center',
   'menu.account.settings': 'Account Settings',
+  'menu.account.setting.modify.settings': 'Settings',
```
Comment (liuxiran) [inline web/src/locales/en-US/menu.ts:61]:
```suggestion
  'menu.account.modifySettings': 'Modify Settings',
```

File: web/src/locales/zh-CN/menu.ts [hunk 3]
```diff
@@ -58,6 +58,7 @@ export default {
   'menu.account': '个人页',
   'menu.account.center': '个人中心',
   'menu.account.settings': '个人设置',
+  'menu.account.setting.modify.settings': '修改设置',
```
Comment (liuxiran) [inline web/src/locales/zh-CN/menu.ts:61]:
```suggestion
  'menu.account.modifySettings': '修改设置',
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 4]
```diff
@@ -133,9 +133,9 @@ const UpstreamForm: React.FC<Props> = forwardRef(
               <Form.Item
                 required
                 key={field.key}
-                label={index === 0 && '节点域名/IP'}
+                label={index === 0 && formatMessage({ id: 'upstream.step.node.domain.or.ip' })}
```
Comment (liuxiran) [inline web/src/components/Upstream/UpstreamForm.tsx:136]: I recommend to name the key refer to:  [basicModule].[moduleName].[elementName].[...desc]

```suggestion
                label={index === 0 && formatMessage({ id: 'page.upstream.form.itemLabel.nodeDomainOrIp' })}
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 5]
```diff
@@ -133,9 +133,9 @@ const UpstreamForm: React.FC<Props> = forwardRef(
...
                 extra={
-                  index === 0 && '使用域名时，默认解析本地 /etc/resolv.conf；权重为0则熔断该节点'
+                  index === 0 && formatMessage({ id: 'upstream.step.domain.name.default.analysis' })
```
Comment (liuxiran) [inline web/src/components/Upstream/UpstreamForm.tsx:138]:
```suggestion
                  index === 0 && formatMessage({ id: 'page.upstream.form.itemExtraMesage.nodeDomainOrIp' })
```

**Comment by liuxiran** [comment]
LGTM except for name style of the i18n key, it would be better to use a unified, meaningful and readable way. 

maybe #331 is a good reference ^^

**Review by LiteSun — CHANGES_REQUESTED** [review]

**Comment by juzhiyuan** [comment]
ping @guoqqqi

File: web/src/locales/en-US/menu.ts (hunk 2 above)
Comment (guoqqqi) [reply web/src/locales/en-US/menu.ts]: The little hump writing style doesn't seem to fit here, so I changed it to ‘menu.account.modify-settings': 'Modify Settings'

Comment (juzhiyuan) [reply web/src/locales/en-US/menu.ts]: Please keep using `Settings` here, for most dashboards, `Setting` is used universal.

File: web/src/components/RightContent/AvatarDropdown.tsx [hunk 6]
```diff
@@ -97,25 +98,25 @@ const AvatarDropdown: React.FC<GlobalHeaderRightProps> = ({ menu }) => {
...
-          个人设置
+          {formatMessage({ id: 'menu.account.settings' })}
         </Menu.Item>
       )}
       {menu && <Menu.Divider />}
 
       <Menu.Item key="settings">
         <SettingFilled />
-        修改设置
+        {formatMessage({ id: 'menu.account.modify-settings' })}
```
Comment (juzhiyuan) [inline web/src/components/RightContent/AvatarDropdown.tsx:114]:
```suggestion
        {formatMessage({ id: 'menu.account.settings' })}
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 7]
```diff
@@ -136,9 +136,9 @@ const UpstreamForm: React.FC<Props> = forwardRef(
               <Form.Item
                 required
                 key={field.key}
-                label={index === 0 && '节点域名/IP'}
+                label={index === 0 && formatMessage({ id: 'page.upstream.form.item-label.node.domain.or.ip' })}
```
Comment (juzhiyuan) [inline web/src/components/Upstream/UpstreamForm.tsx:139]:
```suggestion
                label={index === 0 && formatMessage({ id: 'page.upstream.form.item-label.nodeDomainOrIp' })}
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 8]
```diff
@@ -136,9 +136,9 @@ const UpstreamForm: React.FC<Props> = forwardRef(
...
                 extra={
-                  index === 0 && '使用域名时，默认解析本地 /etc/resolv.conf；权重为0则熔断该节点'
+                  index === 0 && formatMessage({ id: 'page.upstream.form.item.extra-message.node.domain.or.ip' })
```
Comment (juzhiyuan) [inline web/src/components/Upstream/UpstreamForm.tsx:141]:
```suggestion
                  index === 0 && formatMessage({ id: 'page.upstream.form.item.extra-message.nodeDomainOrIp' })
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 9]
```diff
@@ -151,7 +151,7 @@ const UpstreamForm: React.FC<Props> = forwardRef(
                       rules={[
                         {
                           required: true,
-                          message: formatMessage({ id: 'upstream.step.input.domain.name.or.ip' }),
+                          message: formatMessage({ id: 'page.upstream.step.input.domain.name.or.ip' }),
```
Comment (juzhiyuan) [inline web/src/components/Upstream/UpstreamForm.tsx:154]:
```suggestion
                          message: formatMessage({ id: 'page.upstream.step.input.domainNameOrIp' }),
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 10]
```diff
@@ -161,7 +161,10 @@ const UpstreamForm: React.FC<Props> = forwardRef(
                         },
                       ]}
                     >
-                      <Input placeholder="域名/IP" disabled={readonly} />
+                      <Input
+                        placeholder={formatMessage({ id: 'page.upstream.step.domain.name.or.ip' })}
```
Comment (juzhiyuan) [inline web/src/components/Upstream/UpstreamForm.tsx:165]:
```suggestion
                        placeholder={formatMessage({ id: 'page.upstream.step.domainNameOrIp' })}
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 11]
```diff
@@ -231,35 +244,35 @@ const UpstreamForm: React.FC<Props> = forwardRef(
...
-                message: formatMessage({ id: 'upstream.step.input.healthy.checks.active.host' }),
+                message: formatMessage({ id: 'page.upstream.step.input.healthy.checks.active.host' }),
               },
               {
                 pattern: new RegExp(
                   /(^([1-9]?\d|1\d{2}|2[0-4]\d|25[0-5])(\.(25[0-5]|1\d{2}|2[0-4]\d|[1-9]?\d)){3}$|^(?![0-9.]+$)([a-zA-Z0-9_-]+)(\.[a-zA-Z0-9_-]+){0,}$)/,
                   'g',
                 ),
-                message: formatMessage({ id: 'upstream.step.domain.name.or.ip.rule' }),
+                message: formatMessage({ id: 'page.upstream.step.domain.name.or.ip.rule' }),
```
Comment (juzhiyuan) [inline web/src/components/Upstream/UpstreamForm.tsx:254]:
```suggestion
                message: formatMessage({ id: 'page.upstream.step.domainNameOrIpRule' }),
```

File: web/src/locales/en-US/menu.ts [hunk 12]
```diff
@@ -58,6 +58,7 @@ export default {
   'menu.account': 'Account',
   'menu.account.center': 'Account Center',
   'menu.account.settings': 'Account Settings',
+  'menu.account.modify-settings': 'Settings',
```
Comment (juzhiyuan) [inline web/src/locales/en-US/menu.ts:61]:
```suggestion
  'menu.account.settings': 'Settings',
```

File: web/src/locales/zh-CN/menu.ts [hunk 13]
```diff
@@ -58,6 +58,7 @@ export default {
   'menu.account': '个人页',
   'menu.account.center': '个人中心',
   'menu.account.settings': '个人设置',
+  'menu.account.modify-settings': '修改设置',
```
Comment (juzhiyuan) [inline web/src/locales/zh-CN/menu.ts:61]:
```suggestion
  'menu.account.settings': '修改设置',
```

**Review by juzhiyuan — APPROVED** [review]

**Comment by juzhiyuan** [comment]
most texts are updated, we could merge this one first IMO.

File: web/src/components/RightContent/AvatarDropdown.tsx [hunk 14]
```diff
@@ -97,25 +98,25 @@ const AvatarDropdown: React.FC<GlobalHeaderRightProps> = ({ menu }) => {
...
+          {formatMessage({ id: 'menu.account.settings' })}
         </Menu.Item>
       )}
       {menu && <Menu.Divider />}
 
       <Menu.Item key="settings">
         <SettingFilled />
-        修改设置
+        {formatMessage({ id: 'menu.account.settings' })}
       </Menu.Item>
```
Comment (liuxiran) [inline web/src/components/RightContent/AvatarDropdown.tsx:115]: duplicate `settings`

File: web/src/pages/Upstream/locales/zh-CN.ts [hunk 15]
```diff
@@ -15,80 +15,89 @@
...
+  'page.upstream.list.operation': '操作',
+  'page.upstream.list.edit': '编辑',
+  'page.upstream.list.confirm.delete': '确定删除该条记录吗？',
+  'page.upstream.list.confirm': '确定',
+  'page.upstream.list.cancel': '取消',
+  'page.upstream.list.delete.successfully': '删除记录成功',
+  'page.upstream.list.delete': '删除',
+  'page.upstream.list': '上游列表',
+  'page.upstream.list.input': '请输入',
+  'page.upstream.list.create': '创建',
```
Comment (liuxiran) [inline web/src/pages/Upstream/locales/zh-CN.ts:102]: We have global keys  `component.global.delete` `component.global.operation` `component.global.name` `component.global.pleaseEnter` `component.global.edit` `component.global.cancel` `component.global.create` etc. we can consider simplifying some definitions ^^

**Review by liuxiran — APPROVED** [review]

**Comment by juzhiyuan** [comment]
I just did those things for this PR:
1. Update some i18n keys and values;
2. Format all web codes;

File: web/src/pages/Upstream/locales/zh-CN.ts (hunk 15 above)
Comment (juzhiyuan) [reply web/src/pages/Upstream/locales/zh-CN.ts:101]: @guoqqqi Please file a new PR after this PR is merged.

**Review by LiteSun — APPROVED** [review]

**Pull request merged**